src/Native/Runtime/EHHelpers.cpp

```diff
@@ -395,8 +395,15 @@ static UIntNative UnwindWriteBarrierToCaller(
     UIntNative sp = pContext->GetSp();
     UIntNative adjustedFaultingIP = *(UIntNative *)sp;
     pContext->SetSp(sp+sizeof(UIntNative)); // pop the stack
-#elif defined(_ARM_) || defined(_ARM64_)
+#elif defined(_ARM64_)
     UIntNative adjustedFaultingIP = pContext->GetLr();
+#elif defined(_ARM_)
+    UIntNative adjustedFaultingIP = pContext->GetLr();
+    if (pContext->GetIp() == (unsigned int)RhpLockCmpXchg64AVLocation)
```

Comment:
`(unsigned int)RhpLockCmpXchg64AVLocation` -> `(UIntNative)&RhpLockCmpXchg64AVLocation` to match the rest